tensorflow_datasets/image/coil100.py

```diff
@@ -0,0 +1,84 @@
+"""Dataset class for COIL-100 dataset"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import fnmatch
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+_URL = "http://www.cs.columbia.edu/CAVE/databases/SLAM_coil-20_coil-100/coil-100/coil-100.zip"
+
+_DESCRIPTION = ("""The dataset contains 7200 color images of 100 objects
+(72 images per object). The objects have a wide variety of complex geometric and reflectance characteristics.
+The objects were placed on a motorized turntable against a black background.
+The turntable was rotated through 360 degrees to vary object pose with respect to a fxed color camera.
+Images of the objects were taken at pose intervals of	5 degrees.This corresponds to
+72 poses per object""")
+
+_LABELS = [str(x) for x in range(0, 360, 5)]
+
+_IMAGE_SHAPE = (128, 128, 3)
+
+_CITATION = """\
+@article{nene1996columbia,
+  title={Columbia object image library (coil-20)},
+  author={Nene, Sameer A and Nayar, Shree K and Murase, Hiroshi and others},
+  year={1996},
+  publisher={Technical report CUCS-005-96}
+}
+"""
+
+class Coil100(tfds.core.GeneratorBasedBuilder):
+  """COIL-100 Image Dataset Class"""
+
+  VERSION = tfds.core.Version('1.0.0')
+
+  UNSTABLE = "Unable to download on secured networks(eg. University Network)"
+
+  def _info(self):
+    """Define Dataset Info"""
+
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=(_DESCRIPTION),
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=_IMAGE_SHAPE),
+            "label": tfds.features.ClassLabel(names=_LABELS),
+            "object_id": tfds.features.Text()
```

Comment:
Shouldn't the object_id label be the class instead ?
People training model on this dataset probably want to predict the object class from the image.
The orientation should be given as is as integer.

So you would have something like:

```python
{
            "image": tfds.features.Image(shape=_IMAGE_SHAPE),
            "label": tfds.features.ClassLabel(names=['obj{}'.format(i) for i in range(1, 101)]),
            "orientation": tf.int64,
}
```